### PR TITLE
Add details regarding abstract and virtual functions in traits

### PR DESCRIPTION
**Pull request by Oni-giri**
Hi!

I noticed that the docs didn't include any mention of `virtual` and `abstract` functions in traits, which can be overridden in the implementation contract. The virtual keyword for functions can be seen in an example in the [Tact by Example](https://tact-by-example.org/05-your-own-trait) website, for instance.

This is a great feature of Tact, so I think it's worth it to include it in the official docs! I also added references to it in other pages which were relevant.

### CODE REVIEW

File: pages/book/contracts.mdx [hunk 1]
```diff
@@ -194,7 +194,7 @@ contract HelloWorld {
 
 These functions behave similarly to private methods in popular object-oriented languages — they're internal to contracts and can be called by prefixing them with a special [identifier `self{:tact}`](#field-access). That's why internal functions can sometimes be referred to as "contract methods".
 
-Internal functions can access the contract's [persistent state variables](#variables) and [constants](#constants).
+Internal functions can access the contract's [persistent state variables](#variables) and [constants](#constants). If declared in a [traits](/book/types#traits), they can be declared as [virtual or abstract](/book/functions#virtual-and-abstract-functions) and overriden in the contract inheriting it.
```
Comment (anton-trunov) [inline pages/book/contracts.mdx:197]: This should be moved to the contract [structure](https://docs.tact-lang.org/book/contracts#structure) subsection. There is this sentence there:

> Furthermore, contracts can inherit all the declarations from [traits](https://docs.tact-lang.org/book/types#traits) and override some of their default behaviours.

File: pages/book/functions.mdx [hunk 2]
```diff
@@ -23,6 +23,34 @@ fun pow(a: Int, c: Int): Int {
...
+
+You can allow the contract inheriting a [traits](/book/types#traits) to modify an internal function, if it has the `virtual` keyword, using `override`. The function can be also marked as `abstract`, in which case the inheriting contract has to define its implementation:
+
+```tact
+trait FilterTrait with Ownable {
+    virtual fun filterMessage(): Bool { // virtual functions can be overridden by users of this trait
+        if (sender() == self.owner) {
+            return false;
+        }
+        return true;
```
Comment (anton-trunov) [inline pages/book/functions.mdx:36]: The following way to express the same meaning is idiomatic:
```suggestion
        return sender() != self.owner;
```

**Review by anton-trunov — CHANGES_REQUESTED** [review]

**Comment by anton-trunov** [comment]
@Oni-giri Thanks for your contribution! Let's fix a couple of nits and we are good to go

**Review by anton-trunov — DISMISSED** [review]

**Comment by anton-trunov** [comment]
@novusnota check my recent changes please

**Review by novusnota — APPROVED** [review]
Looks nice, will be helpful towards #131

**Pull request merged**